### PR TITLE
RBP: Add Pi specific deinterlace support reporting

### PR DESCRIPTION
**Pull request by popcornmix**

### CODE REVIEW

**Comment by popcornmix** [comment]
Does this look right?
I made ResetVideoCodecInfo overridable but I'm not sure if you would prefer a different way of setting these.

File: xbmc/cores/VideoPlayer/Process/overrides/rbpi/ProcessInfoPi.cpp [hunk 1]
```diff
@@ -33,17 +33,33 @@ CProcessInfo* CProcessInfo::CreateInstance()
 // base class definitions
 CProcessInfoPi::CProcessInfoPi()
 {
+}
 
+void CProcessInfoPi::ResetVideoCodecInfo()
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/Process/overrides/rbpi/ProcessInfoPi.cpp]: this method is not supposed to be overridden

File: xbmc/cores/VideoPlayer/Process/overrides/rbpi/ProcessInfoPi.cpp [hunk 2]
```diff
@@ -33,17 +33,33 @@ CProcessInfo* CProcessInfo::CreateInstance()
...
+  deintMethods.push_back(EINTERLACEMETHOD::VS_INTERLACEMETHOD_MMAL_BOB_HALF);
+  UpdateDeinterlacingMethods(deintMethods);
 }
 
 CProcessInfoPi::~CProcessInfoPi()
 {
 
 }
 
+void CProcessInfoPi::SetSwDeinterlacingMethods()
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/Process/overrides/rbpi/ProcessInfoPi.cpp]: this one should set the methods

**Comment by popcornmix** [comment]
So SetSwDeinterlacingMethods is suitable for omxplayer and mmal?
I was assuming "Sw" related to ffmpeg decode.

**Comment by FernetMenta** [comment]
Yes, "Sw" relates to ffmpeg. Platforms may want to add i.e. RENDER_BOB or BLEND, whatever renderer supports. Other codecs than ffmpeg are supposed to call UpdateDeinterlacingMethods() on CProcessInfo to set their own methods.

**Comment by popcornmix** [comment]
Updated - any better?

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp [hunk 3]
```diff
@@ -247,6 +247,15 @@ enum AVPixelFormat CDVDVideoCodecFFmpeg::GetFormat( struct AVCodecContext * avct
       {
         ctx->m_processInfo.SetVideoPixelFormat(pixFmtName ? pixFmtName : "");
         ctx->SetHardware(dec);
+
+        std::list<EINTERLACEMETHOD> deintMethods;
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp]: why this? ffmpeg should not be touched. you can override SetSwDeinterlacingMethods, if you want to add methods like render_bob.

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp]: ah, this is for mmal. then it should be done in mmal. the decoder has the processInfo instance.

Comment (popcornmix) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp]: This path is for MMALFFMPEG - the zero copy ffmpeg acceleration.
That doesn't currently have access to processInfo.
MMALCodec (full hardware decode) does do it directly.

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp]: why don't you pass the processInfo object in the constructor like we do for other decoders?

**Comment by popcornmix** [comment]
Updated to use processInfo inside MMALFFMPEG.

**Comment by FernetMenta** [comment]
thanks

**Pull request merged**